src/Catalyst.Core.Lib.Tests/UnitTests/Cryptography/CmsTest.cs

```diff
@@ -0,0 +1,178 @@
+igg5jozKCW82JsuWSiW9tu0F/6DuvYiZwHS3OLiJP0CuLfbOaRw8Jia1RTvXEH7m
+3N0/kZ8hJIK4M/t/UAlALjeNtFxYrFgsPgLxxcq7al1ruG7zBq8L/G3RnkSjtHqE
+cn4oisOvxCprs4aM9UVjtZTCjfyNpX8UWwT1W3rySV+KQNhxuMy3RzmL
+-----END ENCRYPTED PRIVATE KEY-----
+";
+            var key = await _keyStoreService.ImportAsync("alice", alice, "mypassword".ToArray());
+            try
+            {
+                Assert.Equal(aliceKid, key.Id);
+
+                var cipher = Convert.FromBase64String(@"
+MIIBcwYJKoZIhvcNAQcDoIIBZDCCAWACAQAxgfowgfcCAQAwYDBbMQ0wCwYDVQQK
+EwRpcGZzMREwDwYDVQQLEwhrZXlzdG9yZTE3MDUGA1UEAxMuUW1OekJxUHdwNDJI
+WkpjY3NMdGM0b2s2TGpaQXNwY2tnczJkdTV0VG1qUGZGQQIBATANBgkqhkiG9w0B
+AQEFAASBgLKXCZQYmMLuQ8m0Ex/rr3KNK+Q2+QG1zIbIQ9MFPUNQ7AOgGOHyL40k
+d1gr188EHuiwd90PafZoQF9VRSX9YtwGNqAE8+LD8VaITxCFbLGRTjAqeOUHR8cO
+knU1yykWGkdlbclCuu0NaAfmb8o0OX50CbEKZB7xmsv8tnqn0H0jMF4GCSqGSIb3
+DQEHATAdBglghkgBZQMEASoEEP/PW1JWehQx6/dsLkp/Mf+gMgQwFM9liLTqC56B
+nHILFmhac/+a/StQOKuf9dx5qXeGvt9LnwKuGGSfNX4g+dTkoa6N
+");
+                var plain = await _keyStoreService.ReadProtectedDataAsync(cipher);
+                var plainText = Encoding.UTF8.GetString(plain);
+                Assert.Equal("This is a message from Alice to Bob", plainText);
+            }
+            finally
+            {
+                await _keyStoreService.RemoveAsync("alice");
+            }
+        }
+
+        [Fact]
+        public void ReadCms_FailsWithoutKey()
+        {
+            var cipher = Convert.FromBase64String(@"
+MIIBcwYJKoZIhvcNAQcDoIIBZDCCAWACAQAxgfowgfcCAQAwYDBbMQ0wCwYDVQQK
+EwRpcGZzMREwDwYDVQQLEwhrZXlzdG9yZTE3MDUGA1UEAxMuUW1OekJxUHdwNDJI
+WkpjY3NMdGM0b2s2TGpaQXNwY2tnczJkdTV0VG1qUGZGQQIBATANBgkqhkiG9w0B
+AQEFAASBgLKXCZQYmMLuQ8m0Ex/rr3KNK+Q2+QG1zIbIQ9MFPUNQ7AOgGOHyL40k
+d1gr188EHuiwd90PafZoQF9VRSX9YtwGNqAE8+LD8VaITxCFbLGRTjAqeOUHR8cO
+knU1yykWGkdlbclCuu0NaAfmb8o0OX50CbEKZB7xmsv8tnqn0H0jMF4GCSqGSIb3
+DQEHATAdBglghkgBZQMEASoEEP/PW1JWehQx6/dsLkp/Mf+gMgQwFM9liLTqC56B
+nHILFmhac/+a/StQOKuf9dx5qXeGvt9LnwKuGGSfNX4g+dTkoa6N
+");
+            ExceptionAssert.Throws<KeyNotFoundException>(() =>
+            {
+                var _ = _keyStoreService.ReadProtectedDataAsync(cipher).Result;
+            });
+        }
+
+        [Fact]
+        public async Task CreateCms_Rsa()
+        {
+            var _ = await _keyStoreService.CreateAsync("alice", "rsa", 512);
+            try
+            {
+                var data = new byte[] {1, 2, 3, 4};
+                var cipher = await _keyStoreService.CreateProtectedDataAsync("alice", data);
+                var plain = await _keyStoreService.ReadProtectedDataAsync(cipher);
+                Assert.Equal(data, plain);
+            }
+            finally
+            {
+                await _keyStoreService.RemoveAsync("alice");
+            }
+        }
+
+        [Fact]
+        public async Task CreateCms_Secp256k1()
+        {
+            var _ = await _keyStoreService.CreateAsync("alice", "secp256k1", 0);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)